### PR TITLE
pr

### PR DESCRIPTION
**Pull request by shinyurim**
- 관리자 카테고리 추가 / 수정 / 삭제 기능
- 의뢰게시판 글목록 확인(글번호, 제목, 조회수, 의뢰금액, 의뢰기간)
- 의뢰게시판 글쓰기 기능(로그인 유저에 한함)

### CODE REVIEW

**Comment by shinyurim** [comment]
|File|Coverage [3.93%]|:x:|
|:-|:-:|:-:|
|[SecurityConfiguration.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FMember%2Fconfiguration%2FSecurityConfiguration.java)|100%|:green_apple:|
|[AdminMainController.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FController%2FAdminMainController.java)|60%|:x:|
|[productController.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2Fproduct%2Fcontroller%2FproductController.java)|60%|:x:|
|[CustomLoginFailureHandler.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FMember%2Fconfiguration%2FCustomLoginFailureHandler.java)|26.92%|:x:|
|[boardServiceImpl.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FrequestBoard%2Fservice%2FboardServiceImpl.java)|25%|:x:|
|[CustomLoginSuccessHandler.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FMember%2Fconfiguration%2FCustomLoginSuccessHandler.java)|11.67%|:x:|
|[requestController.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FrequestBoard%2Fcontroller%2FrequestController.java)|11.11%|:x:|
|[AdminMemberController.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FController%2FAdminMemberController.java)|7.83%|:x:|
|[BuyerServiceImpl.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FMember%2FService%2FBuyerServiceImpl.java)|7.74%|:x:|
|[CategoryServiceimpl.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FService%2FCategoryServiceimpl.java)|7.5%|:x:|
|[SellerServiceImpl.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FMember%2FService%2FSellerServiceImpl.java)|6.09%|:x:|
|[Category.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FEntity%2FCategory.java)|1.14%|:x:|
|[requestBoard.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FrequestBoard%2Fentity%2FrequestBoard.java)|0.69%|:x:|
|[Admin.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FEntity%2FAdmin.java)|0%|:x:|
|[Product.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2Fproduct%2Fentity%2FProduct.java)|0%|:x:|
|[BoardDto.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FrequestBoard%2Fdto%2FBoardDto.java)|0%|:x:|
|[CategoryDto.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FDto%2FCategoryDto.java)|0%|:x:|
|[SellerDto.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FDto%2FSellerDto.java)|0%|:x:|
|[BuyerDto.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2FDto%2FBuyerDto.java)|0%|:x:|
|[SellerParam.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2Fmodel%2FSellerParam.java)|0%|:x:|
|[BuyerParam.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2Fmodel%2FBuyerParam.java)|0%|:x:|
|[CategoryInput.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2FAdmin%2Fmodel%2FCategoryInput.java)|0%|:x:|
|[PageUtil.java](https://github.com/shinyurim/TadeM/blob/5883d1ddf0157a9cf844e268c9916008372cf584/src%2Fmain%2Fjava%2Fproject%2FTadeM%2Futil%2FPageUtil.java)|0%|:x:|

|Total Project Coverage|3.19%|:x:|
|:-|:-:|:-:|

File: src/main/java/project/TadeM/Admin/Dto/CategoryDto.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+@Builder
+public class CategoryDto {
+
+	Long id;
+	String categoryName;
+	int sortValue;
+	boolean usingYn;
+
+	public static List<CategoryDto> of(List<Category> categories) { // 목록 구현
+		if (categories != null) {
```
Comment (KimHyoJin) [inline src/main/java/project/TadeM/Admin/Dto/CategoryDto.java:24]: `categories.stream().map(Objects::nonNull).map(CategoryDto::of).collect(Collectors.toList())
`
로 대체해볼수 있을 것 같아요

File: src/main/java/project/TadeM/Admin/Entity/Category.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@NoArgsConstructor
+public class Category {
+
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	Long id;
+
+	String categoryName;
+	int sortValue;
+	boolean usingYn;
```
Comment (KimHyoJin) [inline src/main/java/project/TadeM/Admin/Entity/Category.java:25]: boolean이라는 필드타입으로 t/f가 온다는 것을 알 수 있기때문에 using정도의 naming도 괜찮을 것 같습니다

File: src/main/java/project/TadeM/Admin/Service/CategoryServiceimpl.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    }
+
+    @Override
+    public List<CategoryDto> list() {
+        List<Category> categories = categoryRepository.findAll(getSortBySortValueDesc());
+        return CategoryDto.of(categories);
+    }
+
+    @Override
+    public boolean add(String categoryName) {
```
Comment (KimHyoJin) [inline src/main/java/project/TadeM/Admin/Service/CategoryServiceimpl.java:32]: 성공시에는 void, 실패시에는 exception을 던지도록 하는 방법도 좋을 것 같습니다. 현재 로직에서는 false를 던지는 부분이 없기도 하구요, 실패시에 false를 던지면 어떤이유에서 실패했는지 응답을 받는쪽에서 알기가 어렵습니다

File: src/main/java/project/TadeM/Admin/Service/CategoryServiceimpl.java [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                .build();
+        categoryRepository.save(category);
+
+        return true;
+    }
+
+    @Override
+    public boolean update(CategoryInput parameter) {
+
+        Optional<Category> optionalCategory = categoryRepository.findById(parameter.getId());
```
Comment (KimHyoJin) [inline src/main/java/project/TadeM/Admin/Service/CategoryServiceimpl.java:47]: category가 존재하지 않을경우는 error를 던지지 않아도 될까요?

**Review by KimHyoJin — APPROVED** [review]
안녕하세요! 밍밍멘토입니다. 간단히 몇가지 리뷰드렸습니다. 수고 많으셨어요!

**Pull request closed without merging**